### PR TITLE
Make file drag and drop in terminal editor work

### LINKED ISSUE #156390: Terminal Editor file drag and drop tried to open file instead of pasting path
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.69.2
- OS Version: MacOS BigSur and Monterey

Steps to Reproduce:

1.  Open a terminal editor.
2. Drag a file from Finder into the terminal.  This tries to open the file (unlike when the terminal is in the panel, which will paste the path).

### PR DESCRIPTION
**Pull request by ssigwart**
Fixes #156390

This makes it so drag and drop events always go to terminal editors to match the way it is handled in the bottom panel.  To test, you can drag a file into the terminal editor and it should paste the file path.

### CODE REVIEW

**Comment by ssigwart** [comment]
@Tyriar, I updated this to fix the failed check.  Also, I noticed that it worked spottily, so I added an additional check in `EditorDropTarget`.

File: src/vs/workbench/common/editor.ts [hunk 1]
```diff
@@ -738,7 +738,13 @@ export const enum EditorInputCapabilities {
 	 * Signals that the editor is composed of multiple editors
 	 * within.
 	 */
-	MultipleEditors = 1 << 8
+	MultipleEditors = 1 << 8,
+
+	/**
+	 * Signals that the editor is should always handle dropping
```
Comment (meganrogge) [inline src/vs/workbench/common/editor.ts:744]:
```suggestion
	 * Signals that the editor should always handle dropping
```

**Review by meganrogge — APPROVED** [review]
Works 👍🏼

**Comment by ssigwart** [comment]
Thanks, @meganrogge.

**Review by Tyriar — CHANGES_REQUESTED** [review]
@mjbvz this PR seems to replace dnd into a terminal editor with putting the file in. Any code pointers for how we could integrate with the shift+dnd overlay mechanism so we can still keep both behaviors?

**Review by mjbvz — CHANGES_REQUESTED** [review]
Can you use the existing `CanDropIntoEditor` capability instead? This lets users choose if they open the file in the editor group (the default), or hold down shift to drop the file into the terminal?

**Comment by ssigwart** [comment]
I believe the current stable version works fine with `CanDropIntoEditor` and the `editor.dropIntoEditor.enabled` setting. However, the panel version of the terminal always pastes the path while the editor version only pastes the path when shift is held. Otherwise, it opens the file.

This PR makes it so they behave consistently so the editor always pastes too.

However, if you want to have an option for it, my opinion is that it should be separate from `editor.dropIntoEditor.enabled`. I'd like a setting to say to always paste into the terminal while still being able to open files when dropped in any other type of editor.

**Comment by mjbvz** [comment]
@ssigwart Looks like terminal editors do not set that flag today. After adding it:

```diff
diff --git a/src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts b/src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts
index dd5684d1ad8..4641dc903af 100644
--- a/src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts
+++ b/src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts
@@ -56,7 +56,7 @@ export class TerminalEditorInput extends EditorInput implements IEditorCloseHand
 	}
 
 	override get capabilities(): EditorInputCapabilities {
-		return EditorInputCapabilities.Readonly | EditorInputCapabilities.Singleton;
+		return EditorInputCapabilities.Readonly | EditorInputCapabilities.Singleton | EditorInputCapabilities.CanDropIntoEditor;
 	}
 
 	setTerminalInstance(instance: ITerminalInstance): void {
```

You can hold shift to drop into a terminal editor. If you drop without holding shift it opens the file as a new editor. That seems like the expected behavior to me, since it aligns dropping into terminal editors with dropping into normal editors

Want to update this PR to use `EditorInputCapabilities.CanDropIntoEditor` instead?

**Comment by Tyriar** [comment]
Thanks for the contribution @ssigwart, I created https://github.com/microsoft/vscode/pull/168611 with @mjbvz's suggestion and it works great

**Comment by ssigwart** [comment]
Thanks, @Tyriar and @mjbvz.

**Pull request closed without merging**